### ISSUE TITLE
failure to ship datapoints do not have a useful link to the actual data

### ISSUE THREAD

**Issue by bzbarsky** [GoogleChrome/confluence#170]
If I click a "failure to ship" graph datapoint and try to look at what APIs are claimed to be represented by that datapoint, I get an empty list.

If I click "expand all", the list stays empty.  After some more random clicking, I can sometimes get data.

**Comment by othermaciej** [comment]
I think this is a performance problem, not functional. But sometimes it takes a very long time.

**Comment by mdittmer** [comment]
Some of these data points pull data from a large number of browsers. We have some ideas in the works for compressing the data model (which is, atm, a little verbose), but launching that change will take time.

If you're not seeing the little progress bar above the data table while you wait for data, that's a bug that can probably be fixed more quickly (but you'll still have to wait a while for the data :sad:)

**Comment by foolip** [comment]
@mdittmer, maybe add a toast that says to hold on after a few seconds of loading?

**Comment by othermaciej** [comment]
I didn't get that the subtly pulsing blue bar is supposed to be a progress bar (assuming that's what it is). It's not very obvious.

**Comment by bzbarsky** [comment]
> I didn't get that the subtly pulsing blue bar is supposed to be a progress bar

Oh.  I hadn't even noticed its existence.

I think a much more obvious progress indicator is in order if the primary UI is still being loaded. ;)

**Comment by othermaciej** [comment]
> Oh. I hadn't even noticed its existence.

I didn't notice it at all either until someone claimed in this issue that there was a progress bar. Without having seen this issue, I'd probably still be unaware of it.

**Comment by foolip** [comment]
Added the roadmap label to this. @mdittmer, what do you think is the most productive way to fix this, by improving the load time or making the loading more obvious? Will there always be some cases that take >1s to load?

**Comment by bzbarsky** [comment]
For what it's worth, 1s would be pretty awesome.  Right now I get a 5-10s lag after clicking "expand all" before anything shows up.  And I'm on a reasonably fast (50Mbit) connection.

**Comment by foolip** [comment]
Yep. I think it should be possible from a raw data volume point of view, because there are about 6000 APIs, so once the list of API names is known, it should be about 750 bytes of data per browser uncompressed.

**Comment by mdittmer** [comment]
My thoughts are:

ASAP, add spinner in addition to progress bar. Progress bar will be the
only thing visible when data is rendered but being updated (e.g., fetching
an unfetched release's data).


On the roadmap: compress data model and make it feasible to have all data
loaded in a format that is much closer to the view's way of presenting it.
